src/test/java/com/icthh/xm/ms/dashboard/web/rest/AtomicBulkDashboardResourceIntTest.java

```diff
@@ -0,0 +1,136 @@
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+import static org.springframework.test.web.servlet.setup.MockMvcBuilders.standaloneSetup;
+
+@RunWith(SpringRunner.class)
+@WithMockUser(authorities = "SUPER-ADMIN")
+@SpringBootTest(classes = {DashboardApp.class, SecurityBeanOverrideConfiguration.class})
+public class AtomicBulkDashboardResourceIntTest {
+
+    private final ObjectMapper mapper = new ObjectMapper();
+
+    private MockMvc httpMock;
+    @Autowired
+    private DashboardRepository dashboardRepository;
+    @Autowired
+    private ExceptionTranslator exceptionTranslator;
+    @Autowired
+    private BulkDashboardResource dashboardResource;
+    @Autowired
+    private MappingJackson2HttpMessageConverter jacksonMessageConverter;
+    @Autowired
+    private PageableHandlerMethodArgumentResolver pageableArgumentResolver;
+
+    @Before
+    public void setup() {
+        initMocks(this);
+        httpMock = standaloneSetup(dashboardResource)
+            .setCustomArgumentResolvers(pageableArgumentResolver)
+            .setControllerAdvice(exceptionTranslator)
+            .setMessageConverters(jacksonMessageConverter)
+            .build();
+    }
+
+    @Before
+    @SneakyThrows
+    public void databaseSetup() {
+        dashboardRepository.saveAll(
+            asList(
+                mapper.readValue(readAsString("dashboardEntity.json"), Dashboard.class),
+                mapper.readValue(readAsString("dashboardEntity.json"), Dashboard.class),
+
+                mapper.readValue(readAsString("dashboardEntity.json"), Dashboard.class),
+                mapper.readValue(readAsString("dashboardEntity.json"), Dashboard.class),
+
+                mapper.readValue(readAsString("dashboardEntity.json"), Dashboard.class),
+
+                mapper.readValue(readAsString("dashboardEntity.json"), Dashboard.class)
+            )
+        );
+    }
+
+    @Test
+    @SneakyThrows
+    public void shouldAtomicCreateDashboards() {
+        httpMock.perform(post("/api/dashboards/bulk")
+            .contentType(APPLICATION_JSON)
+            .param("isAtomicProcessing", "true")
+            .content(readAsString("bulkAtomicCreateDashboards.json")))
+            .andExpect(status().isOk());
+    }
+
+    @Test
+    @SneakyThrows
+    public void shouldAtomicUpdateDashboards() {
+        httpMock.perform(put("/api/dashboards/bulk")
+            .contentType(APPLICATION_JSON)
+            .param("isAtomicProcessing", "true")
+            .content(readAsString("bulkAtomicUpdateDashboards.json")))
+            .andExpect(status().isOk());
+    }
+
+    @Test
+    @SneakyThrows
+    public void shouldAtomicDeleteDashboards() {
+        httpMock.perform(delete("/api/dashboards/bulk")
+            .contentType(APPLICATION_JSON)
+            .content(readAsString("bulkAtomicDeleteDashboards.json")))
+            .andExpect(status().isOk());
+
+        assertNulls(953L, 954L);
```

Comment:
do not use direct IDs as they may change

Reply:
exactly, this is the purpose of tests if you change anything it should fail

Reply:
exactly, these ids for my cases in my separate for this test method file. If they being changed then it would be updated here. Plus I do not have the ability as data for test is in the file